lib/logstash/inputs/beats.rb

```diff
@@ -189,6 +205,17 @@ def register
       @logger.warn("configured ssl_key => #{@ssl_key.inspect} will not be used") if @ssl_key
     end
 
+    @enrich = resolve_enriches
+
+    @include_codec_tag = original_params.include?('include_codec_tag') ? params['include_codec_tag'] : include_codec_metadata?
+    @ssl_peer_metadata = original_params.include?('ssl_peer_metadata') ? params['ssl_peer_metadata'] : @enrich.include?('ssl_peer_metadata')
+
+    # intentionally ask users to provide codec when they want to use the codec metadata
+    # second layer enrich is also a controller, provide enrich => ['codec_metadata' or 'default'] with codec if you override event original
+    if !original_params.include?('codec') || (@enrich && (!@enrich.include?('codec_metadata') || @enrich.include?('none')))
```

Comment:
Self note: this is a breaking change! Clarify with @yaauie and suggest raw transformation (`lib/logstash/inputs/beats/message_listener.rb:L43`)